SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -739,6 +767,59 @@ void FSpatialGDKEditorToolbarModule::StopSpatialServiceButtonClicked()
 	});
 }
 
+void FSpatialGDKEditorToolbarModule::ToggleSpatialDebuggerEditor()
+{
+	// Don't try and start spatial editor debugger spatial networking is disabled.
+	if (!GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
+	{
+		UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("Attempted to toggle spatial debugger in editor but spatial networking is disabled."));
+		return;
+	}
+
+	if (SpatialDebuggerEditor != nullptr)
+	{
+		bSpatialDebuggerEditorEnabled = !bSpatialDebuggerEditorEnabled;
+
+		USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+		SpatialGDKEditorSettings->SetSpatialDebuggerEditorEnabled(bSpatialDebuggerEditorEnabled);
+
+		SpatialDebuggerEditor->ShowWorkerRegions(bSpatialDebuggerEditorEnabled);
+	}
+	else
+	{
+		UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("There was no SpatialDebuggerEditor setup when the map was loaded."));
+	}
+
+}
+
+
+void FSpatialGDKEditorToolbarModule::MapChanged(UWorld* World, EMapChangeType MapChangeType)
+{
+	if (MapChangeType == EMapChangeType::LoadMap || MapChangeType == EMapChangeType::NewMap)
+	{
+		// If spatial networking is enabled then create the spatial debugger for this map
+		if (GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
+		{
+			// Create the spatial debugger for this map
+			UWorld* EditorWorld = GEditor->GetEditorWorldContext().World();
+			check(EditorWorld);
+
+			FActorSpawnParameters spawnParameters;
+			spawnParameters.bHideFromSceneOutliner = true;
+			SpatialDebuggerEditor = EditorWorld->SpawnActor<ASpatialDebuggerEditor>(spawnParameters);
```

Comment:
pascal case for vars```suggestion
			FActorSpawnParameters SpawnParameters;
			SpawnParameters.bHideFromSceneOutliner = true;
			SpatialDebuggerEditor = EditorWorld->SpawnActor<ASpatialDebuggerEditor>(SpawnParameters);
```